File: Java/easy/P_665.java
package easy;

public class P_665 {

    public boolean checkPossibility(int[] nums) {
        int cnt = 0;
        for (int i = 1; i < nums.length && cnt <= 1; i++) {
            if (nums[i - 1] > nums[i]) {
                cnt++;
                if (i - 2 < 0 || nums[i - 2] <= nums[i]) {
                    nums[i - 1] = nums[i];
                } else {
                    nums[i] = nums[i - 1];
                }
            }
        }
        return cnt <= 1;
    }
}

File: cpp/leetcode/P_665.cpp
#include <bits/stdc++.h>
#define fast_io                \
  ios::sync_with_stdio(false); \
  cin.tie(nullptr);
using namespace std;

bool checkPossibility(vector<int>& nums) {
  int n = nums.size();
  for (int i = 0; i < n - 1; i++) {
    if (nums[i] > nums[i + 1]) {
      int t = nums[i + 1];
      nums[i + 1] = nums[i];
      if (isIncreasing(nums, n)) {
        return true;
      }
      nums[i] = nums[i + 1] = t;
      return isIncreasing(nums, n);
    }
  }
  return true;
}

bool isIncreasing(vector<int>& nums, int n) {
  for (int i = 0; i < n - 1; i++) {
    if (nums[i] > nums[i + 1]) {
      return false;
    }
  }
  return true;
}
